packages/automerge-repo/src/synchronizer/CollectionSynchronizer.ts

```diff
@@ -85,6 +85,18 @@ export class CollectionSynchronizer extends Synchronizer {
     )
 
     const documentId = message.documentId
+    const senderId = message.senderId
+    const okToSync = await this.repo.syncPolicy(senderId, documentId)
```

Comment:
was casually observing Discord and took a look at the PR, pardon my intrusion, just wanted to mention that `await somePromise()` will always put a new event into the event loop **at the very end**, thus even an innocent no-op (like `async () => true`) can create a lot of latency if the event loop is busy/saturated. So it is possible that it might be better to make it optional and thus not `await`ing as the default behavior.